cf/samples/javascript-samples/geotab/geotab-to-sap-cloud.js

```diff
@@ -0,0 +1,768 @@
+const DEBUG_ITERATIONS_LIMIT = 20;
+// The maximum number of records to fetch from a Geotab database per poll request.
+const MAX_RECORDS_PER_ITERATION = 100;
+
+/** 
+ * Data types known by SAP IoT Service.
+ * @enum {string}
+ */
+const DataType = {
+    INTEGER: 'integer', 
+    LONG: 'long', 
+    FLOAT: 'float', 
+    DOUBLE: 'double',
+    BOOLEAN: 'boolean', 
+    STRING: 'string', 
+    BINARY: 'binary', 
+    DATE: 'date'
+}
+
+// Read the configuration file.
+const config = readConfigFile('config.json');
+
+/**
+ * An instance of the Geotab API.
+ * @type {Object}
+ */
+const geotab = new Geotab_API(config.Geotab_username, config.Geotab_password, config.Geotab_database);
+
+/** 
+ * The URL of the Swagger (OpenAPI 2.0) specification of the SAP IoT Service northbound API
+ * @type {string}
+ */
+const SAP_Swagger_API = `https://${config.SAP_instance}.${config.SAP_landscape}/${config.SAP_instance}/iot/core/api/v1/doc/swagger/api`;
+
+/**
+ * A proxy object for posting measurement data to SAP IoT Service.
+ * 
+ * The IoT Service data model allows for creating router devices that are authenticated once and then
+ * forward the data on behalf of other devices.
+ * @type {Object}
+ */
+let routerDevice;
+
+/** 
+ * An instance of the SAP IoT Service northbound API.
+ * 
+ * The instance is created with the user credentials and the tenant ID, and it allows for creating devices, sensors, etc.
+ * for the corresponding SAP IoT Service tenant.
+ * @type {Object}
+ */
+let SAP_DeviceManagement;
+
+/** 
+ * The ID of the SAP IoT Services Cloud REST Gateway.
+ * @type {string}
+ */
+let gatewayId;
+
+// IoT Services data types to which the corresponding Geotab typeNames are mapped.
+// More types like these can be created by the programmer.
+/** 
+ * The ID of the SAP sensor type corresponding to the Geotab typeName 'LogRecord'.
+ * @type {string}
+ */
+let sensorTypeId_LogRecord;
+/** 
+ * The ID of the SAP sensor type corresponding to the Geotab typeName 'StatusData'.
+ * @type {string}
+ */
+let sensorTypeId_StatusData;
+
+// More caches like these can be created on demand
+/** 
+ * A cache mapping Geotab device IDs to Vehichle Identification Numbers (VIN)
+ * @const {Object<string, string>}
+ */
+const geotabDeviceCache = {};
+/** 
+ * A cache mapping Geotab Diagnostic IDs to objects describing the Diagnostics
+ * @const {Object<string, Object>}
+ */
+const geotabDiagnosticCache = {}; // key is Geotab Diagnostic ID, value is a dictionary of properties
+
+main();
+
+// Only function declarations after this point.
+
+/**
+ * The main function.
+ */
+async function main() {
+    SAP_DeviceManagement = await new Swagger(SAP_Swagger_API, {
+        authorizations: {
+            basicAuth: {
+                username: config.SAP_username,
+                password: config.SAP_password
+            }
+        }
+    });
+    //console.log('SAP northbound API', SAP_DeviceManagement);
```

Comment:
remove commented code, or add a comment in case it's required

Reply:
Fixed here and in all similar cases.